src/definitions/README.md

```diff
@@ -18,7 +18,7 @@ The key is the string defined in the rippled source code, such as "LedgerEntry",
 
 ### nth
 
-`nth` is the index (sort code) used to construct the field code (Field ID) of this field, for use in ordering it with other fields of the same data type.
+`nth` is the index (field code) used to construct the field code (Field ID) of this field, for use in ordering it with other fields of the same data type.
```

Comment:
In the xrpl.org docs, it refers to something called a "sort code":

> For example, the Account field of a Payment transaction has sort code 1 , so it comes before the Destination field which has sort code 3 .

https://xrpl.org/serialization.html#field-codes

Since `Account` has "nth": 1 and `Destination` has "nth": 3, I think "nth" is the sort code, isn't it?

Then, the sort code is [combined](https://github.com/ripple/rippled/blob/eaff9a0e6aec0ad077f118501791c7684debcfd5/src/ripple/protocol/SField.h#L95-L98) with the type code to construct the field code with `(id << 16) | index`?

I could be wrong; this is just what I'm getting from reading the docs. If it's better to call the `nth` the field code, then maybe we should update the xrpl.org docs to change the reference to a "sort code".

Reply:
nth means `nth of type` yeah and is its `ordinal`

Reply:
> type code to construct the field code with (id << 16) | index?

I'm guessing the `Field ID` is the (type.code << 16 | field.nth) number.
I think this lib used to refer to it as a field "ordinal"

Reply:
Looks to me that the field ID is computed differently from the ordinal, and is used for sorting the Fields. Here is how field ID is encoded in rippled https://github.com/seelabs/rippled/blob/cecc0ad75849a1d50cc573188ad301ca65519a5b/src/ripple/protocol/impl/Serializer.cpp#L117-L148

Reply:
Sort code or field code both sound like the same thing to me.

Reply:
OK. Then "field code" is the term that I have confused. Would the following be accurate?

> `nth` is the field code, which is combined with the type code in order to construct the Field ID of this field. The Field ID is only used for sorting the fields. Since there are multiple fields with the same data type, the `nth` (meaning "nth of type") is used to deterministically order each field among other fields of the same data type.